demo/src/test/java/com/mongodb/jdbc/utils/PrintUtils.java

```diff
@@ -0,0 +1,188 @@
+                row[2] = rsMeta.getTableName(i);
+                row[3] = rsMeta.getColumnName(i);
+                row[4] = String.valueOf(rsMeta.getColumnType(i));
+                row[5] = rsMeta.getColumnTypeName(i);
+                row[6] = String.valueOf(rsMeta.getColumnDisplaySize(i));
+                row[7] = String.valueOf(rsMeta.getPrecision(i));
+                row[8] = String.valueOf(rsMeta.getScale(i));
+                row[9] = rsMeta.getColumnClassName(i);
+                row[10] = rsMeta.getColumnLabel(i);
+
+                while (hasMoreData) {
+                    hasMoreData = false;
+                    for (int j = 0; j < row.length; j++) {
+                        String data = row[j];
+                        int cutIndex = Math.min(data.length(), MAX_RS_META_COL_WIDTH);
+                        sb.append(String.format(getFormat(MAX_RS_META_COL_WIDTH), data.substring(0, cutIndex)));
+                        if (cutIndex == MAX_RS_META_COL_WIDTH) {
+                            row[j] = data.substring(cutIndex);
+                            hasMoreData = true;
+                        }
+                        else
+                        {
+                            row[j] = "";
+                        }
+                    }
+                    sb.append("|\n");
+                }
+            }
+            printRowSeparator(sb, colNames.size(), MAX_RS_META_COL_WIDTH);
+
+            System.out.println(sb.toString());
+        }
+        catch (Exception e)
+        {
+            e.printStackTrace();
+            throw e;
+        }
+    }
+
+    public static void printResultSet(ResultSet rs) throws SQLException
+    {
+        try {
+            // Get the metadata of this rs.
+            ResultSetMetaData rsMeta;
+            rsMeta = rs.getMetaData();
+
+            int columnCount = rsMeta.getColumnCount();
+
+            StringBuilder sb = new StringBuilder();
+
+            int[] maxColsWidth = printRsHeader(sb, columnCount, rsMeta);
+            printRsContent(sb, columnCount, rs, maxColsWidth);
+
+            System.out.println(sb.toString());
+        }
+        catch (Exception e)
+        {
+            e.printStackTrace();
+            throw e;
+        }
+    }
+
+    private static String getFormat(int maxColWidth) throws SQLException
+    {
+        return "| %-" + maxColWidth + "s ";
+    }
+
+    private static void printRsContent(StringBuilder sb, int columnCount, ResultSet rs, int[] maxColsWidth) throws SQLException
+    {
+        String data = null;
+        while (rs.next()) {
+            String[] row = new String[columnCount];
+            boolean hasMoreData;
+            do {
+                hasMoreData = false;
+                for (int i = 0; i < columnCount; i++) {
+                    if (row[i] == null) {
+                        row[i] = rs.getString(i + 1);
+                        if (row[i] == null ) {row[i] = NULL_STR;}
+                    }
```

Comment:
I think we can use a ternary here
```suggestion
                    if (row[i] == null) {
                        String rowValue = rs.getString(i + 1);
                        row[i] = rowValue == null ? NULL_STR : rowValue;
                    }
```

Reply:
I figured a ternary would be better because it seems a little odd to me that we're checking `if (row[i] == null)` twice within two lines of one another. Ternaries are also idiomatic in Java, right? If you really think it makes things hard to read, then we can keep it as is, but I do think the current code looks a little awkward.

Reply:
You seem to really like to have a ternary here. If this makes you happy, then it's Ok for me too. I can live with it. It's not a big deal either way. For me, reading the if statement is easier for me than the "duplicated" rowValue and having to backtrack on the ==. Plus, the simple if is also actually what we want to do => only replace the value of row[i] if it's null.
But ternary operator are wonderful for replacing simple if/else. Things like `max = (n1 > n2) ? n1 : n2`